src/Nethermind/Nethermind.Blockchain.Test/ReadOnlyBlockTreeTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using FluentAssertions;
+using Nethermind.Core.Test.Builders;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using NUnit.Framework;
+
+namespace Nethermind.Blockchain.Test
+{
+    public class ReadOnlyBlockTreeTests
+    {
+        private IBlockTree _innerBlockTree;
+        private ReadOnlyBlockTree _blockTree;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _innerBlockTree = Substitute.For<IBlockTree>();
+            _blockTree = new ReadOnlyBlockTree(_innerBlockTree);            
+        }
+        
+        [Test]
+        public void DeleteChainSlice_throws_when_endNumber_specified()
```

Comment:
_not_specified?

Reply:
what does the name of the test mean?

Reply:
externally we don't support endNumber (well actually we shouldn't support endNumber other than BestKnown), to not corrupt the chain